### PR TITLE
fix issue 187:If only 1 main class detected, proceed and start it dir…

### PR DESCRIPTION
**Pull request by andxu**
…ectly

### CODE REVIEW

File: src/configurationProvider.ts [hunk 1]
```diff
@@ -134,7 +134,9 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                     }).sort ((a, b): number => {
                         return a.label > b.label ? 1 : -1;
                     });
-                    const selection = await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" });
+                    const selection = pickItems.length > 1 ?
+                        await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" })
```
Comment (yaohaizh) [inline src/configurationProvider.ts]: Select main class<project name> ?? Not understand which should be selected? Both?

File: src/configurationProvider.ts [hunk 2]
```diff
@@ -134,7 +134,9 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                     }).sort ((a, b): number => {
                         return a.label > b.label ? 1 : -1;
                     });
-                    const selection = await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" });
+                    const selection = pickItems.length > 1 ?
+                        await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" })
+                        : pickItems[0];
```
Comment (testforstephen) [inline src/configurationProvider.ts]: If there is no main class in the workspace, what happens here?

Comment (andxu) [reply src/configurationProvider.ts]: early exit at the top of this method.

File: src/configurationProvider.ts (hunk 1 above)
Comment (andxu) [reply src/configurationProvider.ts]: yes, if it is not specified in launch.json, java debug will list them for the choosing from user, if only one main class is found, I will use it directly without popup windows for one chosen option.

File: src/configurationProvider.ts (hunk 2 above)
Comment (testforstephen) [reply src/configurationProvider.ts]: Please extract the logic to a method, e.g. `promptForMainClass`

File: src/configurationProvider.ts [hunk 3]
```diff
@@ -262,3 +238,43 @@ function convertLogLevel(commonLogLevel: string) {
...
     }
 }
+
+interface IMainClassOption {
+    readonly projectName?: string;
+    readonly mainClass?: string;
+}
+
+async function chooseMainClass(folder: vscode.WorkspaceFolder | undefined): Promise<IMainClassOption> {
+    const res = <any[]>(await resolveMainClass(folder ? folder.uri : undefined));
```
Comment (testforstephen) [inline src/configurationProvider.ts]: `any[]` => `IMainClassOption[]`

File: src/configurationProvider.ts [hunk 4]
```diff
@@ -262,3 +238,43 @@ function convertLogLevel(commonLogLevel: string) {
             return "FINE";
     }
 }
+
+interface IMainClassOption {
+    readonly projectName?: string;
+    readonly mainClass?: string;
```
Comment (testforstephen) [inline src/configurationProvider.ts]: Can `mainClass` be empty?

Comment (andxu) [reply src/configurationProvider.ts]: as a model, I don't want to forbid mainClass to null

**Pull request merged**